util/patch/patch.go

```diff
@@ -230,7 +230,7 @@ func (h *Helper) patchStatusConditions(ctx context.Context, obj client.Object, f
 		}
 
 		// Create the condition patch before merging conditions.
-		conditionsPatch := client.MergeFromWithOptions(latest.DeepCopyObject(), client.MergeFromWithOptimisticLock{})
+		conditionsPatch := client.MergeFromWithOptions(latest.DeepCopyObject().(conditions.Setter), client.MergeFromWithOptimisticLock{})
```

Comment:
For my knowledge, what is the reason for this changes, it seems kind of unreleated 😅

Reply:
Controller Runtime accepts an object here, not a runtime one anymore